### PR TITLE
fix enable_fips_mode remediations

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- modify Bash and Ansible remediations of enable_fips_mode
- apart from running fips-mode-setup --enable, it now applies also the cryptopolicy configured in the profile.

#### Rationale:

- the previous implementation only used the fips-mode-setup utility to enable the FIPS mode. However, this utility always configures the "fips" cryptopolicy. However, some profiles (such as OSPP), uses a policy based on "fips" - "fips:ospp". This policy is based on "fips", it is even stricter. But due to hardcoded configuration of "fips" policy done by fips-mode-setup, the remediation always resulted in error.
- This implementation should not cause problems even if the profile for some reason tries to use FIPS mode and at the same time it tries to use policy not based on fips. In that case, the check test_system_crypto_policy_value will always make rule fail.

### CODE REVIEW

**Comment by vojtapolasek** [comment]
I am aware that I used copy paste. We can consider macro.

File: linux_os/guide/system/software/integrity/fips/enable_fips_mode/ansible/shared.yml [hunk 1]
```diff
@@ -13,3 +13,13 @@
   command: /usr/bin/fips-mode-setup --enable
   when:
     - is_fips_enabled.stdout.find('FIPS mode is enabled.') == -1
+
+- name: "{{{ rule_title }}}"
+  lineinfile:
+    path: /etc/crypto-policies/config
+    regexp: '^(?!#)(\S+)$'
+    line: "{{ var_system_crypto_policy }}"
```
Comment (yuumasato) [inline linux_os/guide/system/software/integrity/fips/enable_fips_mode/ansible/shared.yml:21]: I believe you need to instantiate `var_system_crypto_policy`

**Review by yuumasato — COMMENTED** [review]
I'd also update the rule description and mention that the crypto policy is expected to match `var_system_crypto_policy`

**Comment by vojtapolasek** [comment]
I have updated the remediation and rule description. I was thinking about having there some Jinja conditional based on the value of var_system_crypto_policy... something like if the policy is set to FIPS, there will be no additional sentence. If there is a different policy, the sentence mentioning that the rule additionally configures a cryptography policy  would be added. But I am not sure if the build system actually supports different rule description for different profiles.

File: linux_os/guide/system/software/integrity/fips/enable_fips_mode/rule.yml [hunk 2]
```diff
@@ -50,6 +51,10 @@ ocil: |-
     <pre>fips-mode-setup --check</pre>
     The output should contain the following:
     <pre>FIPS mode is enabled.</pre>
+    To verify that cryptography policy has been configured correctly, run the
```
Comment (yuumasato) [inline linux_os/guide/system/software/integrity/fips/enable_fips_mode/rule.yml:54]:
```suggestion
    To verify that the cryptographic policy has been configured correctly, run the
```

**Review by yuumasato — CHANGES_REQUESTED** [review]

**Comment by yuumasato** [comment]
> I was thinking about having there some Jinja conditional based on the value of var_system_crypto_policy... something like if the policy is set to FIPS, there will be no additional sentence. If there is a different policy, the sentence mentioning that the rule additionally configures a cryptography policy would be added. But I am not sure if the build system actually supports different rule description for different profiles.

I don't think this is possible with XCCDF.
The Jinja macros and conditionals are processed during content build time.
And the dynamicity of the rule descriptions at rule evaluation time are provided by `xccdf:sub` which only provides simple text substitution. AFAIK XCCDF doesn't support complex processing in the rule descriptions.

**Comment by vojtapolasek** [comment]
I have accepted your suggestion. HOwever, I noticed that the variable does not show in ocil. However, it is displayed correctly in the rule description. Is that expected?

**Comment by yuumasato** [comment]
> I have accepted your suggestion. HOwever, I noticed that the variable does not show in ocil. However, it is displayed correctly in the rule description. Is that expected?

Kind of :(
https://github.com/ComplianceAsCode/content/issues/4354

**Review by yuumasato — APPROVED** [review]

**Comment by yuumasato** [comment]
Seems like `ci/prow/e2e-aws-rhcos4-moderate` is failing on infra.

**Pull request merged**